packages/datepicker/src/react/calendar.tsx

```diff
@@ -1,162 +1,186 @@
+import { css, keyframes } from 'glamor'
+
+import {
+  HTMLPropsFor,
+  RefFor,
+  ValueOf
+} from '@pluralsight/ps-design-system-util'
+
+import { RenderProps } from 'dayzed'
+
 import {
   CaretLeftIcon,
   CaretRightIcon
 } from '@pluralsight/ps-design-system-icon'
-import Theme from '@pluralsight/ps-design-system-theme'
-import { compose, css, keyframes } from 'glamor'
-import React, { useRef, useState, useEffect, FC } from 'react'
+
+import { DateContext } from './context'
 
 import stylesheet from '../css'
-import {
-  arrayOf,
-  getDaysInMonth,
-  getNextMonthYear,
-  getPrevMonthYear,
-  getMonthName,
-  getFirstDayOfWeekForMonth
-} from '../js'
+import { slides } from '../vars'
 
-const slide = keyframes(
-  stylesheet['@keyframes psds-date-picker__calendar__keyframes__slide']
+const forward = keyframes(
+  stylesheet['@keyframes psds-calendar__keyframes__forward']
+)
+const backward = keyframes(
+  stylesheet['@keyframes psds-calendar__keyframes__backward']
 )
 
 const styles = {
-  calendar: () => css(stylesheet['.psds-date-picker__calendar']({ slide })),
-  days: () => css(stylesheet['.psds-date-picker__calendar__days']),
-  day: (isSelected: boolean) =>
-    compose(
-      css(stylesheet['.psds-date-picker__calendar__day']),
-      isSelected &&
-        css(stylesheet['.psds-date-picker__calendar__day--selected'])
+  calendar: () => css(stylesheet['.psds-calendar']),
+  headerWrapper: () => css(stylesheet[`.psds-calendar__header-wrapper`]),
+  gridWrapper: () => css(stylesheet[`.psds-calendar__grid-wrapper`]),
+  gridSlide: (slide?: ValueOf<typeof slides>) =>
+    css(
+      stylesheet[`.psds-calendar__grid-slide`],
+      slide === 'forward' &&
+        stylesheet[`.psds-calendar__grid-slide--forward`](forward),
+      slide === 'backward' &&
+        stylesheet[`.psds-calendar__grid-slide--backward`](backward)
     ),
-  overlay: () => css(stylesheet['.psds-date-picker__overlay']),
-  skippedDay: () => css(stylesheet['.psds-date-picker__calendar__skipped-day']),
-  switcher: () => css(stylesheet['.psds-date-picker__calendar__switcher']),
-  switcherMonth: () =>
-    css(stylesheet['.psds-date-picker__calendar__switcher__month']),
-  weekHeading: () =>
-    css(stylesheet['.psds-date-picker__calendar__week-heading']),
-  weekHeadingDay: () =>
-    css(stylesheet['.psds-date-picker__calendar__week-heading__day'])
+  month: () => css(stylesheet['.psds-calendar__month']),
+  header: () => css(stylesheet['.psds-calendar__header']),
+  headerButton: () => css(stylesheet['.psds-calendar__header-button']),
+  headerMonth: () => css(stylesheet['.psds-calendar__header-month']),
+  weekdayHeader: () => css(stylesheet['.psds-calendar__weekday-header']),
+  dateFiller: () => css(stylesheet['.psds-calendar__filler']),
+  dateGrid: () => css(stylesheet['.psds-calendar__date-grid'])
 }
 
-interface CalendarProps {
-  value: Date | undefined
-  onSelect: (evt: React.MouseEvent, date: Date) => void
+const monthNamesShort = [
```

Comment:
Month names are available through js i18n. Looks to be IE11 compatible. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleString  Maybe week names too?

Reply:
This won't work but not a bad idea to make the calendar support multiple locales dynamically